crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -659,20 +659,8 @@ end
 #! - note_idx is the index of the note to which the asset is added.
 #! - ASSET can be a fungible or non-fungible asset.
 #!
-#! Panics if:
-#! - the procedure is called from a non-account context.
-#! - the invocation of this procedure does not originate from the native account.
-#!
 #! Invocation: dynexec
 export.note_add_asset
-    # check that this procedure was executed against the native account
-    exec.memory::assert_native_account
-    # => [note_idx, ASSET, pad(11)]
```

Comment:
Why is removing this OK? Would we not want to make sure that these methods are executed only for the native account?

Reply:
We want to allow transaction and note scripts to call these functions:
- creating an empty note doesn't affect the account state
- `add_asset_to_note` by itself doesn't touch account state either, it needs to be combined with `accout::remove_asset`.  But, exposing it to a note script lets that note script create an output in-flight SWAP note, where assets are not required to "pass through" the account. (See [this comment](https://github.com/0xMiden/miden-base/issues/1333#issuecomment-2886230289) for a motivating in-flight swap example, and instead of having a new component `wallet_plus` we directly expose the method)

Let me know if this helps!

Reply:
Yeah, this makes sense. However, for foreign accounts do we still want to allow calling `tx_create_note`? Feels like it should not be possible but I'm probably missing something

Reply:
Resolution: for now bring back native account assert, and create a new issue to discuss for the future whether removing native assert can be useful for some context (e.g. service fees)

Reply:
I think imposing the restriction on foreign accounts is fine, but I'm not sure this solves the issue. It would still be possible to use the faucet as an anonymizer, it is just that the calls to create the anonymized note would be made from a note being consumed.

I think this would need to be addressed in the authentication procedure of the faucet. I think requiring a signature for output notes would solve this - but there may be other solutions too.